Add more examples for pyiceberg view (#3414)
apache/iceberg-python: mkdocs/docs/api.md

cleanup_old_snapshots("analytics.user_events", [12345, 67890, 11111])
```

## Create a view

To create a view from a catalog:

```python
import time
import pyarrow as pa
from pyiceberg.catalog import load_catalog
from pyiceberg.view import SQLViewRepresentation, ViewVersion

> catalog = load_catalog("default")

Reviewer: I would add a section on top of create view to specify that in order to use the view endpoints from the catalog, the view configuration needs to have `"view-endpoints-supported": "true"`

```python
from pyiceberg.catalog import load_catalog

catalog = load_catalog(
    "docs",
    **{
        "uri": "http://127.0.0.1:8181",
        "s3.endpoint": "http://127.0.0.1:9000",
        "py-io-impl": "pyiceberg.io.pyarrow.PyArrowFileIO",
        "s3.access-key-id": "admin",
        "s3.secret-access-key": "password",
        "view-endpoints-supported": "true",
    }
)
```

> identifier = "default.some_view"

Reviewer: Nit: I would just put the string directly in the create_view call

> schema = pa.schema([pa.field("some_col", pa.int32())])

Reviewer: Nit: for the example I believe would be best to show an Iceberg schema type instead of arrow but also note that create_view also accepts an arrow schema

view_version = ViewVersion(
    version_id=1,
    schema_id=1,
    timestamp_ms=int(time.time() * 1000),

>     summary={},

Reviewer: ```suggestion
    summary={"spark-version": "4.1"},
```

Reviewer: Just to show what the summary is usually used for

Reviewer 2: Using "spark" seems weird. That is a different query engine. 

Note that common metadata keys are `engine-name` and `engine-version`, not `spark-version`. https://iceberg.apache.org/view-spec/#versions

Reviewer 2: Gentle reminder.

Author: Thanks for the reminder. Will update this later tonight.

Author: This is fixed now @ebyhr . Again, thanks for the review.

cleanup_old_snapshots("analytics.user_events", [12345, 67890, 11111])
```

## Create a view

> If the REST server does not indicate support for view endpoints, you can enable it by setting `"view-endpoints-supported": "true"`:

Reviewer 3: I don't think we should be encouraging this pattern. If a Catalog says they don't support view endpoints (through /v1/config), we shouldn't tell people to ignore that.

Reviewer 2: @rambleraptor, what is your suggestion? This config property was introduced for backward compatibility. Older REST servers may not send `endpoints` field in the ConfigResponse even if view operations are supported.

Reviewer 4: Would it make sense to explicitly call this out as the intended purpose?
```suggestion
Older REST server implementations sometimes do not advertise support for view endpoints. If yours supports views but doesn't advertise it, you can specify this and enable view functionality by setting `"view-endpoints-supported": "true"`:
```

Author: @rambleraptor @ebyhr what do yours guys think?

Reviewer 3: If PyIceberg is unable to automatically determine view support on your REST Catalog, you can manually specify, `"view-endpoints-supported": "true"`:
```

How does that sound?

Author: @rambleraptor I like this. I will wait for tomorrow before pushing above in case there are better wording.

```python
from pyiceberg.catalog import load_catalog

catalog = load_catalog(
    "docs",
    **{
        "uri": "http://127.0.0.1:8181",
        "s3.endpoint": "http://127.0.0.1:9000",
        "py-io-impl": "pyiceberg.io.pyarrow.PyArrowFileIO",
        "s3.access-key-id": "admin",
        "s3.secret-access-key": "password",

>         "view-endpoints-supported": "true",

Reviewer 2: Documenting `view-endpoints-supported` is a good idea. However, placing it under `Create a view` seems unusual because other view operations also require it in older REST catalog implementations.

How about keeping `## Views` and mentioning this snippet there? Each section can be a H3 `###`. 
Or, we could add the property to `configuration.md`.

Author: So the create/load/check tables are all H2. Thus, I proceeded with the same for views. I like the idea of keeping `## Views`, however, I am not sure if we should make the rest H3 as they will be a bit off compare to the table ones. WDYT?

## Views

If PyIceberg is unable to automatically determine view support on your REST Catalog, you can manually specify, `"view-endpoints-supported": "true"`:

```python
from pyiceberg.catalog import load_catalog

catalog = load_catalog(
    "docs",
    **{
        "uri": "http://127.0.0.1:8181",

>         "py-io-impl": "pyiceberg.io.pyarrow.PyArrowFileIO",

Reviewer 2: > py-io-impl

This is actually an optional parameter for views, right?

Author: Yes. But the [top page](https://github.com/MonkeyCanCode/iceberg-python/blob/main/mkdocs/docs/api.md#python-api) also use this, do we not want to be consistent with the doc?

        "py-io-impl": "pyiceberg.io.pyarrow.PyArrowFileIO",
        "s3.access-key-id": "admin",

>         "s3.secret-access-key": "password",

Reviewer 2: nit: We could move the `s3.endpoint` so that we can place S3 properties close together:
```suggestion
        "py-io-impl": "pyiceberg.io.pyarrow.PyArrowFileIO",
        "s3.endpoint": "http://127.0.0.1:9000",
        "s3.access-key-id": "admin",
        "s3.secret-access-key": "password",
```

        "view-endpoints-supported": "true",
    }
)
```

## Create a view

To create a view from the catalog:

```python
import time
from pyiceberg.catalog import load_catalog
from pyiceberg.schema import Schema
from pyiceberg.types import IntegerType, NestedField
from pyiceberg.view import SQLViewRepresentation, ViewVersion

catalog = load_catalog("default")

schema = Schema(NestedField(field_id=1, name="some_col", field_type=IntegerType(), required=False))
view_version = ViewVersion(
    version_id=1,
    schema_id=1,
    timestamp_ms=int(time.time() * 1000),

>     summary={"engine-name": "spark", "engineVersion": "4.1"},

Reviewer 2: nit: the summary field is expected to store the metadata created the object:
```suggestion
    summary={"engine-name": "pyiceberg", "engine-version": "0.11.1"},
```

>     summary={"engine-name": "pyiceberg", "engine-version": "0.11.1"},

Reviewer 2: We've merged https://github.com/apache/iceberg-python/pull/3458 setting some fields by default. `version_id` and `timestamp_ms` lines can be removed from this example. 

`summary` also can be removed once #3441 lands.

Author: Thanks for the headup. This ha been removed to use the default along with un-used import (`time).
